src/pip/_internal/utils/unpacking.py

```diff
@@ -225,9 +232,7 @@ def untar_file(filename, location):
                 tar.utime(member, path)  # type: ignore
                 # member have any execute permissions for user/group/world?
                 if member.mode & 0o111:
-                    # make dest file have execute for user/group/world
-                    # no-op on windows per python docs
-                    os.chmod(path, (0o777 - current_umask() | 0o111))
+                    set_file_modes(path)
```

Comment:
I would not extract that in a function, if only to avoid scratching our head to find a good name for it (set_file_modes is not accurate).

Reply:
I did that as per the comment https://github.com/pypa/pip/issues/8164#issuecomment-621011719 . 

What would be an appropriate name for this function btw?

Reply:
The accurate name to this operation would need be something like `set_extracted_file_to_default_mode_plus_executable` to me.

Reply:
> The accurate name to this operation would need be something like `set_extracted_file_to_default_mode_plus_executable` to me.

That's indeed a better name. Updated in https://github.com/pypa/pip/pull/8375/commits/a8e20e36b2f75fcae8ec3f176ddc53e333923f29